File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> romToInt = Map.of(
            'I', 1,
            'V', 5,
            'X', 10,
            'L', 50,
            'C', 100,
            'D', 500,
            'M', 1000
        );

        int total = 0;
        int n = s.length();

        for (int i = 0; i < n; i++) {
            int curr = romToInt.get(s.charAt(i));
            int next = (i + 1 < n) ? romToInt.get(s.charAt(i + 1)) : 0;

            if (curr < next) {
                total -= curr;
            } else {
                total += curr;
            }
        }

        return total;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_to_int : Dict[str: int] = {
            'I' : 1,
            'V' : 5,
            'X' : 10,
            'L' : 50,
            'C' : 100,
            'D' : 500,
            'M' : 1000
        }
        total : int = 0
        for i in range(len(s)-1):
            prev : int = roman_to_int[s[i]]
            next : int = roman_to_int[s[i+1]]
            if prev >= next:
                total += prev
            else :
                total -= prev
        total += roman_to_int[s[len(s)-1]]

        return total